strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/security/ldap/support/SpringSecurityLdapInternalsTester.java

```diff
@@ -0,0 +1,105 @@
+package org.carlspring.strongbox.controllers.configuration.security.ldap.support;
+
+import org.carlspring.strongbox.authentication.support.AuthoritiesExternalToInternalMapper;
+import org.carlspring.strongbox.forms.configuration.security.ldap.LdapConfigurationForm;
+import org.carlspring.strongbox.forms.configuration.security.ldap.LdapConfigurationTestForm;
+import org.carlspring.strongbox.forms.configuration.security.ldap.LdapSearchForm;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.ldap.core.ContextSource;
+import org.springframework.ldap.core.support.BaseLdapPathContextSource;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.ldap.authentication.BindAuthenticator;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.security.ldap.authentication.LdapAuthenticator;
+import org.springframework.security.ldap.search.FilterBasedLdapUserSearch;
+import org.springframework.security.ldap.userdetails.DefaultLdapAuthoritiesPopulator;
+import org.springframework.security.ldap.userdetails.LdapAuthoritiesPopulator;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class SpringSecurityLdapInternalsTester
```

Comment:
@fuss86 @steve-todorov how this assumed to be work according our custom authentication implementation? I mean that we assume to have LDAP configuration inside `strongbox-authentication-providers.xml`, so why we need this additional configuration of `LdapAuthenticationProvider` here?

Reply:
see SB-1217 point 3. This is to test the configuration.

Reply:
Yeah, there needs to be an option to test out the ldap configuration settings you've configured from the front-end before you actually apply them. This is necessary so you don't get locked out because of bad/invalid configuration settings (i.e. bad lookup `dn`, wrong group lookup `dn`, `base dn`, etc). Of course if you're manually editing the `strongbox-authentication-providers.xml` you would be skipping the "test" part as it is assumed you know what you're doing.

Reply:
maybe I am missing something but as I see if you will make LDAP configuration changes using `LdapAuthenticatorConfigurationController` then all this changes will be lost on strongbox restart and we will get initial configuration from `strongbox-authentication-providers.xml`.

Reply:
Hmm.. that shouldn't be working like that. If you change the configuration through the `LdapAuthenticatorConfigurationController` it should actually save those settings in the `strongbox-authentication-providers.xml` and reload/refresh the cached `xml` file (I presume once the `xml` file is loaded it's cached or stored in-memory?). @fuss86 can you confirm?

Reply:
Good catch @sbespalov , we need to update `strongbox-authentication-providers.xml` too. I'll do it in this PR. Thanks for your detective skills :)

>  (I presume once the xml file is loaded it's cached or stored in-memory?). @fuss86 can you confirm?

Yes :) I confirm

Reply:
>  we need to update strongbox-authentication-providers.xml too

I think it will be better to store LDAP configuration as our other configurations in `strpngbox.xml` and just use it in Spring managed LDAP components.

Reply:
yeah, for sure I'm not going to reproduce `strongbox-authentication-providers.xml `